src/server/eventHubHelper.ts

```diff
@@ -0,0 +1,83 @@
+
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import { AmqpError, isAmqpError, Connection, ReceiverEvents, parseConnectionString } from "rhea-promise";
+import * as crypto from "crypto";
+
+// The following helper functions are directly copied from:
+// https://raw.githubusercontent.com/Azure/azure-sdk-for-js/main/sdk/eventhub/event-hubs/samples/v5/typescript/src/iothubConnectionString.ts
+export const convertIotHubToEventHubsConnectionString =  async (connectionString: string): Promise<string> =>
+{
+    const { HostName, SharedAccessKeyName, SharedAccessKey } = parseConnectionString<{
+        HostName: string;
+        SharedAccessKeyName: string;
+        SharedAccessKey: string;
+    }>(connectionString);
+  
+    // Verify that the required info is in the connection string.
+    if (!HostName || !SharedAccessKey || !SharedAccessKeyName) {
+        throw new Error(`Invalid IotHub connection string.`);
+    }
+  
+    //Extract the IotHub name from the hostname.
+    const [iotHubName] = HostName.split(".");
+  
+    if (!iotHubName) {
+        throw new Error(`Unable to extract the IotHub name from the connection string.`);
+    }
+  
+    // Generate a token to authenticate to the service.
+    // The code for generateSasToken can be found at https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens
+    const token = generateSasToken(`${HostName}/messages/events`, SharedAccessKey, SharedAccessKeyName, 5);
+  
+    const connection = new Connection({
+        transport: "tls",
+        host: HostName,
+        hostname: HostName,
+        username: `${SharedAccessKeyName}@sas.root.${iotHubName}`,
+        port: 5671,
```

Comment:
Do you think port number should be in a constants file?

Reply:
ah i didn't even notice this, was a pure copy paste. Not sure if it would cause any issue if there is a port conflict, guess we can leave it and figure out?